### PR TITLE
rs: Reed–Solomon virtual backend (alpha)

### PR DESCRIPTION
**Pull request by Breschling**
Adds the `rs` virtual backend (Reed–Solomon over multiple remotes) with `docs/content/rs.md`, fstest/testserver wiring and `test_all` entry, backend `status`/`heal`/`degraded` commands, context-cancel coverage, a 1 GiB integration test, and shell verification scripts (`compare_sequential.sh` local + MinIO).

#### What is the purpose of this change?

This PR adds a new **virtual backend `rs`** that implements Reed–Solomon erasure coding across multiple remotes (k data + m parity) using `github.com/klauspost/reedsolomon`.

Key pieces:

* **Backend behavior**

  * Stripes logical objects across k+m shard remotes with a shared **RCLONERS** footer (v1, 96 bytes, ns `Mtime`).
  * **Read/list at k**; **write quorum** configurable (`write_quorum`, default **k+1**), with compensating rollback when `rollback=true` ([`QUORUM_TRANSACTIONS.md`](backend/rs/docs/QUORUM_TRANSACTIONS.md)).
  * **Virtual padding** on data-shard payloads; streaming upload by default (`use_spooling=false`; spool for unknown-size `Put` only).
  * Optimized read path using data shards when possible; supports range reads via `fs.RangeOption` / `fs.SeekOption`.
  * **Lazy metadata (Nick’s model):** `List` / `NewObject` / copy-move return use shard remote **size** and **ModTime** (1s) when possible — **no footer read on the hot path**; footer lazy for fallback, `Open`, and `Hash` ([`LIST_METADATA.md`](backend/rs/docs/LIST_METADATA.md)).
  * **Cheap `SetModTime`:** quorum `Object.SetModTime` on ModTime-capable shards; footer rewrite only on `ModTimeNotSupported` backends.
  * **Write transactions:** namespace (`Mkdir`, `Rmdir`, `DirMove` + namespace heal) and objects (`Copy`, `Move`, `Remove`, `SetModTime`; `Remove` irreversible once deletes succeed).
  * RS→RS **server-side** `Copy` / `Move` / `DirMove` for compatible layouts.
  * **Heal** reconstructs missing shards; reference ModTime from surviving shard remotes on rebuilt particles.
  * Topology: **`k > m`** enforced.
  * Backend commands:
    * `status`: health / quorum-oriented view of shard remotes.
    * `heal`: heal logical objects by reconstructing and uploading missing shard particles.
      * No argument: scan namespace.
      * Path argument: single-object repair.
      * `-o dry-run=true`: report-only.
    * `degraded`: inspect object/directory skew (`summary`, `ls`, `lsd`).

* **Docs**

  * `docs/content/rs.md` — erasure coding background, shard layout, footer fields, read/write quorum, configuration, range reads, limitations.
  * `backend/rs/docs/LIST_METADATA.md` — normative size/ModTime policy.
  * `backend/rs/docs/QUORUM_TRANSACTIONS.md` — write/namespace transaction spec.
  * `backend/rs/docs/OPEN_QUESTIONS.md` — tracked follow-ups.

* **Tests**

  * `fstest` integration:
    * `fstest/testserver/init.d/TestRsLocal` (4 local shard dirs, **`k=3`, `m=1`**).
    * `backend/rs/rs_fstest_test.go` (`InternalTester` wiring).
    * Entry in `fstest/test_all/config.yaml`.
  * Shell tests in `backend/rs/test` (raid3-style gate):
    * `compare.sh`, `compare_all.sh`, `compare_heal.sh`, **`compare_sequential.sh`** (local + MinIO); `rsverify` for particle checks.
  * Context/cancellation tests in `backend/rs/rs_context_test.go` (Put source read, upload phase, `listAllObjectRemotes` / `heal` listing).
  * Large-file coverage: `TestLargeObjectPutOpen1GiB` (~1 GiB upload/download with MD5 verification).
  * Unit tests: quorum rollback, namespace heal, lazy list/NewObject metadata, SetModTime shard vs footer fallback, heal reference mtime.

`rs` is marked **Alpha / experimental**; follow-up work (large-namespace heal cost, footer-fallback streaming on `ModTimeNotSupported` shards, S3/MinIO quirks, mount eager metadata, auto-heal on list) is tracked in `backend/rs/docs/OPEN_QUESTIONS.md`. `versionIntroduced` is intentionally omitted from `docs/content/rs.md` until the release version that ships `rs` is decided.

#### Updates since the April review comment

Items from the April 13 overview comment on [PR #9301](https://github.com/rclone/rclone/pull/9301):

| Topic | Status |
|-------|--------|
| Spooling off by default; streaming uploads | **Done** |
| Configurable `write_quorum`; `k > m` | **Done** |
| Server-side Copy/Move/DirMove | **Done** |
| Virtual padding (same-name shard grouping) | **Done** (replaces equal-size padding) |
| Footer format documented | **Done** — v1 `RCLONERS`, 96 bytes (replaces draft v3) |
| Rollback | **Done** — quorum transactions + compensating rollback |
| ModTime / sync-friendly metadata | **Done** — shard remote ModTime primary; cheap `SetModTime` |
| Auto-heal on list | **Not implemented** — still for discussion |
| Per-remote metadata beyond ModTime | **Unchanged** — RS footer holds hashes/layout; shard ModTime used for sync |

#### Was the change discussed in an issue or in the forum before?

Discussion: [Erasure coding: planning a Reed–Solomon virtual remote](https://forum.rclone.org/t/erasure-coding-planning-a-reed-solomon-virtual-remote/53538)

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: docs/content/rs.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+`backend/rs/docs/OPEN_QUESTIONS.md` in the source tree.
+
+## How it works
+
+- **Parameters**: `data_shards` = **k**, `parity_shards` = **m**. Configure
+  exactly **k+m** entries in `remotes`, in **shard index order** (shard `0` …
+  shard `k+m-1`).
+- **Encoding**: Content is split into stripes and Reed–Solomon encoded; each
+  shard receives its **particle** (that shard’s payload bytes).
+- **Footer**: Each particle ends with the same **98-byte EC footer layout**
```
Comment (ncw) [inline docs/content/rs.md]: Could you describe the footer (maybe not in this section) in enough detail that you could use it to implement the code? That is what I tried to do in the crypt docs. We want this to be an open standard anyone could implement.

File: backend/rs/docs/OPEN_QUESTIONS.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+This document tracks design gaps, limitations, and follow-up work for the Reed-Solomon (`rs`) virtual backend. When a topic is resolved, note the decision here or in code. **Last reviewed**: 2026-03-28. User-facing overview: [`README.md`](../README.md).
+
+---
+
+## High priority
+
+### Q1: `use_spooling=false` write path
+
+**Status**: Active — not implemented
```
Comment (ncw) [inline backend/rs/docs/OPEN_QUESTIONS.md]: We should be able to do all uploads without a local spool file. I think that should be a design criterion. That is how crypt was designed.

File: backend/rs/docs/OPEN_QUESTIONS.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+**Status**: Active — not implemented  
+**Notes**: `Put` / `PutStream` return an error when `use_spooling` is false (`operations.go`). Streaming encode-and-upload without a full local spool file is still to be designed (memory bounds, stripe buffering, quorum on partial failure).
+
+---
+
+## Medium priority
+
+### Q2: `Fs.Move` / `Fs.Copy` / `DirMove`
+
+**Status**: Active — not exposed on `rs` `Fs`
```
Comment (ncw) [inline backend/rs/docs/OPEN_QUESTIONS.md]: You should be able to move the data about in the directory tree without corrupting it. I think this is the case but just emphasizing it here!

File: backend/rs/docs/OPEN_QUESTIONS.md (hunk 2 above)
Comment (Breschling) [reply backend/rs/docs/OPEN_QUESTIONS.md]: Why I use `_spooling=true`

When I started the RAID3 remote I processed streams directly. Local remotes worked fine, but local MinIO often failed randomly because I used chunked encoding without `Content-Length`. To fix this I introduced spooling so I could compute `Content-Length` and use single‑request uploads. That has been very reliable and reasonably efficient, since encoding and writing the shards as spool files runs in parallel with opening connections to the remotes.

I kept the flag even though only the spooling path exists right now.

Originally I wanted RS without a quorum, i.e. more remotes than `k+m`. In that design, a failed upload to one remote would mean re‑uploading the same shard to a different remote, where having the shard as a spool file is helpful. Now that we rely on a quorum, we no longer strictly need that style of retry.

If we add a streaming path, I see two cases:

1. Input is a file  
   - Without compression: we can compute shard sizes and a fixed footer, derive `Content-Length` upfront and use single‑request PUT.  
   - With compression: we cannot know the final size, so we’d need chunked encoding.

2. Input is a stream (e.g. `rclone rcat`)  
   - Same as “file + compression”: no way to know the final size, so chunked encoding.

From your comment, it sounds like you’d prefer to also implement a streaming path without spooling to disk. We could:

- use single‑request PUT with computed `Content-Length` when possible (file, no compression), and  
- use chunked encoding otherwise.

Would you recommend that approach @ncw?

Comment (ncw) [reply backend/rs/docs/OPEN_QUESTIONS.md]: In rclone the input is always a stream and the backends shouldn't be spooling to disk. Sometimes the size of the stream may be unknown (eg rcat and mount) and it's ok to spool via disk in that case if you have to.

It might work better if you use the OpenChunkWriter interface for uploads - this will have the chunks in rewind able blocks which may be in memory.

How do you split the shards? First 1/n to first file next to second file? Or do you split each rs chunk? Either way streaming should be better possible.

File: backend/rs/docs/OPEN_QUESTIONS.md (hunk 3 above)
Comment (Breschling) [reply backend/rs/docs/OPEN_QUESTIONS.md]: Server-side move/copy probably needs some design discussion.

For RS we have the property that any \(k\) out of \(k+m\) shards are sufficient to reconstruct an object. We currently hardcode a quorum of \(k+1\) on upload to ensure there is at least one shard of redundancy. With, say, \(k=10, m=30\), moving only \(k+1=11\) shards would technically satisfy the quorum, but in practice it creates *copies*: 11 shards are moved, 29 remain, so the same logical object exists in two locations.

One possible implementation may be:

- Restrict to configurations with \(k < m\).  
- Define a server-side move/copy as “successful” if it completes on at least \(k+1\) remotes.  
- Any remotes that were unavailable or missing shards at that time would later need background cleanup to remove stale shards or directories.

Directory-level operations are even trickier. Consider 2 objects on 4 remotes with \(k=2, m=2\):  
- obj A on remotes {1,2,3}  
- obj B on remotes {2,3,4}  

If remote 4 is down and we “move the directory”, the operation can succeed for remotes {1,2,3} but not for 4. When 4 comes back B is broken but A still valid.

Given these corner cases, the simplest semantics may be:

- **Shard move/copy:** require that *all* remotes are available; move/copy whatever shards exist. No partial success semantics.  
- **Directory move:** require that all remotes are available and that the directory contains the same shard set on all remotes; only then allow a server-side directory move.

We may fine tune the semantics later. What do you think?

File: backend/rs/docs/OPEN_QUESTIONS.md (hunk 2 above)
Comment (Breschling) [reply backend/rs/docs/OPEN_QUESTIONS.md]: I’ll add a streaming path and will look into using `OpenChunkWriter`.

For the current spooling path we use “first 1/n to first file, next to second file, …”. And yes, for the streaming path we need to split per RS chunk/stripe.

File: backend/rs/docs/OPEN_QUESTIONS.md (hunk 3 above)
Comment (ncw) [reply backend/rs/docs/OPEN_QUESTIONS.md]: I think that sounds fine to ban certain operations when the backend is in a degraded state...

> We currently hardcode a quorum of (k+1)

I don't think you should be doing this. This will affect the usability. Say you had n=3,k=2,m=1 which would mean you couldn't upload at all if one remote was down. Perhaps this should be user config?

> We currently hardcode a quorum of (k+1) on upload to ensure there is at least one shard of redundancy. With, say, (k=10, m=30), moving only (k+1=11) shards would technically satisfy the quorum, but in practice it creates copies: 11 shards are moved, 29 remain, so the same logical object exists in two locations.

I see what you mean. I used to run a RAID 6 array with 4 disks which is effectively k=2,m=2,n=4. You could take any two disks out and still have all your data. However the two disks you took out could also be used to mount an array. Which is the original array.... Who knows?

A raid controller would mark a disk as needing reconstruction if it had been out the array. We probably don't want do do that if a remote goes offline for a while perhaps that information could be used by heal.

I think it would be OK if the backend became read only at some number of missing remotes ...

**Comment by Breschling** [comment]
Dear @ncw ,

based on your feedback I have revised and extended several aspects of the RS backend. All changes are in the `rs-backend` branch. Below is a short overview; I would appreciate further discussion and review.

Spooling  
Spooling is disabled by default. Uploads (files) and downloads (shards) are handled as streaming pipelines so the RS backend should work correctly for large objects. For `PUT` operations where the caller does not provide a real `ObjectInfo` with size (e.g. `rclone rcat`), the backend falls back to the spooling path to determine the object size before uploading shards.

Parallelism  
Data upload and download operations are mostly executed in parallel. The RS backend currently starts up to `k + m` goroutines per object to process all shards concurrently.

Quorum and k > m constraint  
The default quorum is `k + 1`, but the user can override this. We enforce `k > m` in the backend to make a simple majority rule usable in degraded modes (`m > k` is only rarely used for very special cases). This allows us to treat any directory with fewer than quorum shard files as definitively broken, since we cannot reach a majority of consistent shard sets.

Server-side operations  
We have implemented `Copy`, `Move`, and `DirMove` between RS remotes (RS→RS) using server-side operations when supported by the underlying remotes.

Shard naming and zero padding  
Shard objects use the original file name without an explicit shard index in the key. Reed–Solomon is used with padding, and we append a fixed-size footer so that all shard objects have identical sizes. This enables fast checks such as:  
- Do we have at least quorum shard objects with the same name in the same directory?  
- Do these shard objects all have the same size?  
If both conditions hold, we can inspect the footer, verify hashes, check that all shard indices are present and unique, and then reconstruct the object.

Auto-heal  
The old RAID3 implementation had an `auto_heal` mode. Example: on `ls`, if the directory view meets quorum and one remote is missing the directory, it may be created automatically. I have not reintroduced this yet, but I think it is worth discussing whether something similar is desirable for the RS backend.

modtime handling  
We store the object `modtime` in the footer, since it is critical for `sync`, `check`, etc. I would like to discuss how we should update `modtime` in a backend-friendly way. One possible approach is to use server-side operations and range reads:  
- rename the object,  
- perform a server-side range read of the existing data, and  
- write a new footer with updated `modtime` to the remotes.  
On S3, this might be implemented via multipart uploads and/or server-side copy plus a final footer update.

Per-remote metadata  
Because an RS remote typically wraps a larger set of heterogeneous remotes, supporting arbitrary per-remote metadata does not seem useful. The current design is to keep all RS-specific core metadata in the footer, without relying on per-remote metadata features.

Rollback  
Rollback is not yet implemented in a consistent way. This is still an open item.

Footer format  
I will provide a separate description of the footer layout (fields, versioning, and extensibility). I would be to discuss this format.

File: docs/content/rs.md (hunk 1 above)
Comment (Breschling) [reply docs/content/rs.md]: Dear @ncw ,

here is the rationale behind the current footer design.

#### Header vs footer

The `rs` backend needs to handle streams without knowing size, hashes, or mtime upfront. A footer lets me record all of that after the payload is written, without sidecar metadata objects and their extra failure modes.

#### Performance

Sharding across multiple remotes already adds latency. The metadata format should therefore be cheap: fixed layout, minimal parsing, and at most one range read per shard when verification is needed.

#### Fixed-size footer

Each shard has a fixed-size footer. This allows a cheap shard grouping step based only on names and sizes from `ls`:

1. Find at least `k` shards with the same base name in the same directory across the configured remotes.
2. Check that these candidate shards have the same size.

Only then do we range-read the footers to verify payload length, shard indices/layout, and hashes.

#### Hashes, mtime, and compression

Because `rs` must work over heterogeneous backends with different hash and metadata support, hashes and logical mtime are stored in the footer instead of relying on backend metadata. In the prior work on RAID3 I introduces compression, I used:

- Compressed payload in fixed-size blocks.
- A compact “compression inventory” (block sizes) between payload and footer.
- A footer flag and block count, so range reads can fetch only the compressed blocks that intersect the requested logical range, then decompress and slice.

#### Filenames vs embedded metadata

I initially encoded even/odd as an postfix to the filename for RAID3. However, with `rs` using postfixes leads to ambiguous multiple shard sets of the same object in one directory, for example:

- `remote_A:/somedata/myfile.odt.01`
- `remote_A:/somedata/myfile.odt.02`
- `remote_B:/somedata/myfile.odt.02`
- `remote_B:/somedata/myfile.odt.03`
- `remote_C:/somedata/myfile.odt.03`
- `remote_C:/somedata/myfile.odt.01`

This effectively gives two full copies of the logical object with overlapping shard placements, which complicates listing and deduplication. To avoid that ambiguity, all structural information (indices, layout, hashes) is kept in the footer and filenames stay simple.

I would be happy to discuss whether we should refine or change this format. And yes, I will check the crypt docs and add more detailed documentation so that people can implement the footer themselves.

Comment (Breschling) [reply docs/content/rs.md]: Maybe we could add a short description like the following to https://rclone.org/rc in case `rs` gets added to the rclone remotes. The text also references the `rsverify` command, which I currently use in the shell-based tests.

## RS backend: format of the shard footer

This note describes the **binary trailer** that rclone’s `rs` (Reed–Solomon) backend appends to each **shard particle** stored on the wrapped remotes.

If you access shard remotes **without** going through the `rs` remote, each object is an encoded **fragment** of the logical file plus this footer—not the original file bytes.

### Stored object layout

Each shard stores one object per logical path:

- **Payload**: Reed–Solomon stripe data for that shard (`NumStripes` stripes × `StripeSize` bytes per stripe on this shard).
- **Footer**: a fixed **102-byte** metadata block (described below).

So for a valid particle:

`object_size = NumStripes × StripeSize + 102`

An empty logical file uses **no** payload stripes (`NumStripes = 0`, `StripeSize` may still reflect configuration); the object is then **102 bytes** of footer only.

### EC footer (version 3)

The footer is always exactly **102 bytes** at the **end** of the shard object. All multi-byte integers are **little-endian** unless noted otherwise.

The first bytes are a readable **magic** so tools can recognize rclone RS particles; the rest is binary.

#### Fields (byte offsets 0–101, zero-based)

- **9 bytes** (offsets **0–8**) ASCII magic string `RCLONE/EC` (no NUL terminator).
- **2 bytes** (**9–10**) format version as `uint16`: **3** (current version).
- **8 bytes** (**11–18**) logical content length as `uint64` (the size of the object as seen through the `rs` remote).
- **16 bytes** (**19–34**) MD5 digest of the **logical** (pre-encoding) file content.
- **32 bytes** (**35–66**) SHA-256 digest of the **logical** file content.
- **8 bytes** (**67–74**) modification time as Unix **seconds** since the epoch (`int64` / `uint64` wire format as in rclone’s encoder).
- **4 bytes** (**75–78**) compression tag. Uncompressed particles use **four zero bytes**.
- **4 bytes** (**79–82**) `NumBlocks` as `uint32`. Reserved for a possible future compression layer; **0** in current versions.
- **4 bytes** (**83–86**) algorithm tag. Reed–Solomon particles use the ASCII letters `R` and `S` followed by **two zero bytes** (`RS\0\0`).
- **1 byte** (**87**) `DataShards` (**k**): number of data shards.
- **1 byte** (**88**) `ParityShards` (**m**): number of parity shards.
- **1 byte** (**89**) `CurrentShard`: this shard’s index (**0** … **k+m−1**).
- **4 bytes** (**90–93**) `StripeSize` (**S**) as `uint32`: per-shard bytes appended per RS stripe (the `stripe_fragment_size` in configuration, on the wire as a 32-bit value).
- **4 bytes** (**94–97**) `PayloadCRC32C` as `uint32`: **CRC-32/Castagnoli** over **this shard’s entire payload** (all bytes before the footer). Used to detect corruption of that shard’s stored fragment.
- **4 bytes** (**98–101**) `NumStripes` as `uint32`: number of RS stripes in the payload; payload length is `NumStripes × StripeSize`.

The footer occupies offsets **0…101** inclusive (**102 bytes** total).

### Consistency across shards

For one logical object, the fields that describe the **whole** file—logical size, hashes, mtime, **k**, **m**, algorithm, compression, `NumBlocks`, `StripeSize`, and `NumStripes`—are intended to **match** on every shard’s footer.

Fields that differ **per shard**:

- `CurrentShard` identifies which backend the particle belongs to.
- `PayloadCRC32C` is computed over **that** shard’s payload only.

So footers are **not** bit-identical copies across remotes, even when the logical file is healthy.

### Examples

**Empty logical file** (`ContentLength = 0`, `NumStripes = 0`):

- Payload: **0 bytes**
- Footer: **102 bytes**
- **Total** stored per shard object: **102 bytes**

MD5 and SHA-256 in the footer are the standard digests of **empty** content (same as an empty file on disk).

**Non-empty logical file**:

- Suppose `NumStripes = N` and `StripeSize = S`. Each shard object has **N × S** bytes of payload plus **102** bytes of footer.

Overhead for large objects is **102 bytes per shard**, independent of logical size (the Reed–Solomon expansion is in the payload, not in extra footer bytes).

### Versioning

Rclone currently reads and writes **footer version 3** only. Older versions, if any ever existed in the wild, are not accepted by the current parser.

### Reference implementation

The authoritative layout is in `footer.go` in this backend package (`FooterSize`, `MarshalBinary` / `ParseFooter`, constants `FooterMagic`, `FooterVersion`).

### Inspecting real particles (`rsverify`)

The standalone developer tool **`rsverify`** (`cmd/rsverify`) uses the same `backend/rs` parser. From the repository root: `go build -o rsverify ./cmd/rsverify`.

- **`rsverify footer <particle>`** — reads one shard file and prints decoded footer fields (add **`--json`** for machine-readable output). Fails if magic/CRC do not match a valid particle.
- **`rsverify check <shard1> <shard2> …`** — loads multiple shard files, checks footers and reconstruction, and compares logical MD5/SHA256 to the footer when possible.

Subcommands **`encode`** / **`decode`** help build or strip test particles with the same on-disk layout. See `rsverify --help` and `backend/rs/README.md`.

**Comment by Breschling** [comment]
Follow-up to April/June: list/NewObject use shard remote size+ModTime without eager footer reads; cheap SetModTime; heal/copy-move use the same metadata policy (LIST_METADATA.md). Quorum transactions and rollback are in QUORUM_TRANSACTIONS.md. Size at list time uses virtual padding on data shards. Still for discussion: auto-heal on list; S3/MinIO edge cases (Q14).

**Comment by ncw** [comment]
@Breschling here are some things which need tidying

Can you rebase this please? And also make it pass the copy/move tests (no fallback when shards lack server-side Copy)?

I think TestLargeObjectPutOpen1GiB is too large for general consumption - can you slim it down a bit? It also builds an Fs directly with k=2, m=2, which the backend's own k > m rule forbids, so it tests a configuration users cannot create.

There is spillover from the raid3 branch in the .gitignore and .cursor entries and Raid3 bullet in README.md and maybe other places.

rs needs adding to to docs/content/docs.md, docs/content/_index.md and bin/make_manual.py alongside the other  backends - check the CONTRIBUTING.md for full details

cmd/rsverify is a standalone binary inside cmd/, which is where rclone's subcommands live. I would suggest moving it under backend/rs/test (or making its checks a backend command) so we do not grow a second binary in the main tree.

The overall shape is good and I think this is getting close on design. Happy to keep discussing auto-heal and the range-read verification proposal separately once the branch is green again.

Thank you

Nick

**Comment by Breschling** [comment]
Dear @ncw, the branch is rebased onto current master (twice — at the start and again before this push) and all points from your review are addressed:

Copy/move tests pass (including with shards lacking server-side Copy, e.g. Linux local): Move now uses a Move-only backup+swap sequence when shards have Move but not Copy, and capability gaps surface as ErrorCantCopy/ErrorCantMove up front — no internal fallbacks. Documented in QUORUM_TRANSACTIONS.md.

TestLargeObjectPutOpen1GiB is now 32 MiB + 1 byte with a valid k=3, m=2 geometry, constructed through the validated NewFs path.

raid3 spillover removed from .gitignore, README.md and the test docs.

rs registered in docs/content/docs.md, _index.md, the navbar and bin/make_manual.py.

cmd/rsverify is gone — its checks are now rclone backend verify (CRC/layout always; -o hashes=true for stripe-wise hash verification, -o strict=true to require all k+m shards).

Happy to discuss auto-heal and the range-read verification proposal whenever it suits you. Thank you for the review!

**Comment by Breschling** [comment]
The **lint errors** are in fs/log/log.go:294-295 (SA4023), a file **this branch doesn't touch**. CI installs golangci-lint with version: latest and recently picked up v2.13.1, whose newer staticcheck flags the non-Windows stub of startWindowsEventLog (it always returns an error, so err != nil is "always true" on Linux builds). Current master has the same code, so all PRs should now fail this check until it's fixed upstream.

dsander already reported this: https://github.com/rclone/rclone/pull/9795#issuecomment-5361764582 

All other checks pass, including the previously failing Windows job. Happy to rebase once a fix lands.
